### PR TITLE
[PMF-13] Add FdR function

### PR DESCRIPTION
**Pull request by pasqualespica**
The purpose of this PR is to add a FdR function service to read [nodoInviaFlussoRendicontazione](https://pagopa.atlassian.net/wiki/spaces/PAG/pages/211585234/nodoInviaFlussoRendicontazione), parsing it and publish on pagoPA's event hub as several `json` events

### List of changes

Added : 
  - _Flussi Di Rendicontazione_ `reporting_function.tf`

Updated : 
  - update Fdr apim policy

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/reporting_function.tf [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  name     = format("%s-reporting-fdr-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host reporting-fdr function
+module "reporting_fdr_function_snet" {
+  count                                          = var.reporting_fdr_enabled && var.cidr_subnet_reporting_fdr != null ? 1 : 0
```
Comment (uolter) [inline src/reporting_function.tf:12]: Is it really required this flag **reporting_fdr_enabled** ?

File: src/reporting_function.tf [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "reporting_fdr_function" {
+  count  = var.reporting_fdr_enabled ? 1 : 0
```
Comment (uolter) [inline src/reporting_function.tf:30]: As per my comment above .... do we really really need this flag **reporting_fdr_enabled**.
Why not just create the function in all envs?

**Review by uolter — CHANGES_REQUESTED** [review]

File: src/reporting_function.tf (hunk 1 above)
Comment (pasqualespica) [reply src/reporting_function.tf:12]: no you're right i'll remove it

File: src/reporting_function.tf (hunk 2 above)
Comment (pasqualespica) [reply src/reporting_function.tf:30]: see above response : )

**Review by uolter — APPROVED** [review]

**Pull request merged**